### DISCUSSION TITLE
[Discussion / Documentation] Better "roadmap" visibility and discussion for "coming up"

### DISCUSSION THREAD

**Discussion by rubyFeedback** [oracle/graal#4293] [General] [answered]
Hey there GraalVM Team,

This is more related to documentation and news entries related to GraalVM and any of the associated ecosystem.

Now you lot have a blog here:

https://medium.com/graalvm

The "developer blog". This is ok so far.

Recently a new article was published here titled "GraalVM 22.0 is here!":

https://medium.com/graalvm

Also fine.

Now I am a fairly "new" person in regards to GraalVM; and to some extent Java, though I have
been using Java since almost ~2 years right now, and before that (and still use) ruby for almost
20 years. \o/

What I would love to see is a "roadmap" that is discussed in a little bit more detail every now and
then. Something where we people can say "ok, the GraalVM team is collectively working on 
this, that, or that other major feature in the coming 3-9 months". Something where we kind of 
get an overview of what is happening. I don't mean that everything has to be discussed, mind
you; and I perfectly understand that this is not trivial to plan the future, so I am not suggesting
to add a "crystal ball". What I would rather like to see is if there is some kind of better "news"
reporting - so more like more blog entries, more discussion on what is going to happen next.

Right now I really don't know what is happening. My only "source of information" comes from 
the GraalVM blog, but I feel this misses a few things, e. g. when the main discussions are
mostly about "ok, this is a new release" and then we wait for the next release. In between I
feel a little bit lost.

I will give you a specific example too, so that you can understand from where I am coming
from. (I also gave most of that information view the recent survey, but better to mention it
several times than mention it never at all.)

Currently I am most interested in these aspects:

- ruby + java 

- native-image + polyglot (this in itself is pretty epic, not necessarily due to any speed gain,
though that is cool too, but kind of the "one language to rule them all" idea, as well as
easy-mode distribution of binaries. I could just get the code, use native-image via ruby
scripts on windows, and then distribute that binary to elderly relatives, since that is actually
one big use case for me. I use linux and ruby for literally everything, but for elderly relatives
being able to provide them with helper tools is IMMENSELY useful, also because I can 
invest time fixing their problems in the long run :D  which also leads me to the next aspect)

- GUI tools in general, be it JavaFX, Swing etc... - to have these work on GraalVM would be
super nice

I think these are the three main aspects I am currently interested in. But since my main source
are the dev blog logs, I feel a bit limited here in the sense that I have no real idea if anyone
else is working on something related to that. Anyway, for the TL;DR summary:

**Please consider publishing more "upcoming changes" or discussions towards changes that
may be of some "usefulness" for GraalVM "downstream" users/devs, so that they know what
may be going to happen next. Something like a somewhat more detailed roadmap of 
"up and coming changes", as much as that is feasibly viable to do - not to expect a 100%
accurate "crystal ball prediction", but more like a "we are working on this or that" in some
2-4 months semi-prediction, or something like that.**

**Comment by rodrigar-mx** [comment]
Thanks for contributing to graalvm. We will look at your proposal closely and take it into account.

**Comment by alina-yur** [comment, marked as answer]
hey @rubyFeedback,
thank you for sharing this feedback with us. You are right that our Medium blog is the place where we announce releases and talk about existing features. But that’s only one channel; let me walk you through others, where among other things we talk about upcoming features and changes.
1. [Projects @ GitHub](https://github.com/oracle/graal/projects). Currently we have several projects on GitHub, on which we work ourselves and collaborate with community:
[Native Image](https://github.com/oracle/graal/projects/2)
[Truffle](https://github.com/oracle/graal/projects/4)
[ARM64 support](https://github.com/oracle/graal/projects/6)
[TruffleRuby projects](https://github.com/oracle/truffleruby/projects)
suggestions for adding more projects are welcome.
2. [Feature requests and PRs @ GitHub](https://github.com/oracle/graal/issues?q=+label%3Afeature+). A lot of upcoming changes are discussed and brainstormed in those, like for example see [Implementation of GC phase pause events #4272](https://github.com/oracle/graal/pull/4272). It’s convenient to track specific features this way, like for example [M1 support](https://github.com/oracle/graal/issues/2666), but I agree that is harder to get a high-level picture from those.
3. RFCs. From time to time we publish RFCs to announce upcoming changes and ask for feedback. This was the case for things like Native Image build process & analysis changes, [asking what to prioritize in Native Image](https://mobile.twitter.com/graalvm/status/1440669511774916627), switching to JDK 11 & JDK 17 builds, and [new build output](https://github.com/oracle/graal/issues/3955). These are usually announced on GitHub, [community Slack](https://www.graalvm.org/slack-invitation/) and [Twitter](https://twitter.com/GraalVM).
4. [Community Slack](https://www.graalvm.org/slack-invitation/) and mailing lists (user, announce). Many of changes are announced on those. We try to keep mailing lists low-traffic and only share the most critical announcements t[here](https://github.com/oracle/graal/discussions/3933), Slack is a place for as frequent and detailed discussions as needed.
5. Native Image committer meetings. Those are the place where Native Image committers discuss recent and upcoming changes. See latest notes [here](https://github.com/oracle/graal/discussions/3933).
6. [Advisory Board](https://www.graalvm.org/community/advisory-board/). Every quarter we meet with Advisory Board members to discuss recent and upcoming changes and get their feedback. Meeting notes are available publicly.

I hope those resources help get a better idea of where one can track what’s coming up in GraalVM. We can certainly work more in this direction, like make sure announce upcoming changes on all platforms whenever possible, or add more projects to work on on GitHub. More suggestions are welcome

**Comment by alina-yur** [comment]
Was going through discussions, to give an update on this -- we now also have a roadmap on GitHub: https://github.com/orgs/oracle/projects/6
